csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.shared.cs

```diff
@@ -916,6 +920,32 @@ internal class NativeLib
                                                 out UIntPtr /*(ulong* out)*/ startTime);
         public static DOrtSessionGetProfilingStartTimeNs OrtSessionGetProfilingStartTimeNs;
 
+        [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+        public delegate IntPtr /*(ONNStatus*)*/ DCreateAndRegisterAllocatorV2(
+                                                IntPtr /* (OrtEnv*) */ environment,
+                                                IntPtr /*(char*)*/ provider_type,
+                                                IntPtr /*(OrtMemoryInfo*)*/ mem_info,
+                                                IntPtr /*(OrtArenaCfg*)*/ arena_cfg,
+                                                IntPtr /*(char**)*/ provider_options_keys,
+                                                IntPtr /*(char**)*/ provider_options_values,
+                                                UIntPtr /*(size_t)*/num_keys);
+        public static DCreateAndRegisterAllocatorV2 OrtCreateAndRegisterAllocatorV2;
+
+        [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+        public delegate IntPtr /*(ONNStatus*)*/ DOrtRunAsync(
```

Comment:
Is this intentionally named OrtRunAsync to not collide with the C# RunAsync? How do we make this private since we don't want users to use this API.

Reply:
This is an internal static class, customers are prohibited to instantiate from it.